arangod/RocksDBEngine/ReplicatedRocksDBTransactionState.cpp

```diff
@@ -134,7 +136,9 @@ futures::Future<Result> ReplicatedRocksDBTransactionState::doCommit() {
         for (auto& res : results) {
           auto result = res.get();
           if (result.fail()) {
-            return result;
+            // We finished replication, but failed to commit locally. There is
+            // no going back.
+            FATAL_ERROR_EXIT_CODE(result.errorNumber().value());
```

Comment:
I think we have different exit code which do not agree with error codes. In particular I would suggest to log a precise error statement es well.

Reply:
> I think we have different exit code which do not agree with error codes.
I did not know that, thank you!